pandas/_libs/tslibs/fields.pyx

```diff
@@ -43,6 +48,36 @@ def get_time_micros(ndarray[int64_t] dtindex):
     return micros
 
 
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def get_datetime_isoformats(ndarray[int64_t] dtindex) -> ndarray:
```

Comment:
IIRC cython is not great about mixing and matching annotation formats.  i.e. the `ndarray[int64_t]` is cython-style and the `-> ndarray` is python style and id be (pleasantly) surprised if cython understands that